src/components/SvgIcon.js

```diff
@@ -75,18 +75,39 @@ export function SvgIcon({ name, className = '', inline = false, title = '' }) {
   );
 }
 
+/**
+ * Register an icon for use with the `SvgIcon` component.
+ *
+ * Returns a symbol that can be passed as the `name` prop to `SvgIcon` in order
+ * to render this icon.
+ *
+ * @param {string|symbol} name - A name for this icon
+ * @param {string} markup - SVG markup for the icon
+ * @return {symbol}
+ */
+export function registerIcon(name, markup) {
+  const key = typeof name === 'string' ? Symbol(name) : name;
```

Comment:
Accepting `symbol` here allows for re-registering an icon, as an analog to the `reset` option in `registerIcons`. The need specifically came up in a test: tests were calling `registerIcons` with `reset=true` and there needed to be a way to restore the removed icons.